libmproxy/protocol2/auto.py

```diff
@@ -0,0 +1,16 @@
+from __future__ import (absolute_import, print_function, division, unicode_literals)
+from .layer import Layer
+
+
+class AutoLayer(Layer):
+    def __call__(self):
+        d = self.client_conn.rfile.peek(1)
+        if d[0] == "\x16":
```

Comment:
Fair point. This was really a quick one-off earlier today, there's much more we can do for the heuristics, which we then should document.

Regarding our actual question:
`0x16` signifies a ClientHello, http://www.moserware.com/2009/06/first-few-milliseconds-of-https.html#client-hello